File: BackEnd/Model/UserDatas.js
import mongoose from "mongoose";
import { NotiFications } from "./Notification.js";
import { Review } from "./Review.js";

const UserDatas = new mongoose.Schema({

    _id:{
        type:Number,
        required : true
    },
    id:{
        type:Number,
        required : true
    },
    uname:{
        type:String,
        required:true
    },
    gender:{
        type:String,
        required : true
    },
    name:{
        type:String,
        required:true
    },
    email:{
        type:String,
        required:true
    },
    password:{
        type:String,
        required:true
    },
    region:{
        type:String,
        required : true
    },
    bio:{
        type:String,
        required:true
    },
    admin:{
        type:Boolean,
        required:true
    },
    profilePic:{
        type:{
            url:String
        }
    },
    notification:[
        NotiFications
    ],
    intrestedGenres: {
        philosophy: Boolean,
        action: Boolean,
        fiction: Boolean,
        romance: Boolean,
        history: Boolean,
        nonFiction: Boolean,
        selfHelpBook: Boolean,
        biography: Boolean,
        horror: Boolean,
        politics: Boolean,
        comics: Boolean,
        fantasy: Boolean
    },
    reviews:[Review]

})

export const UserData = mongoose.model("UserData",UserDatas);

File: FrontEnd/src/main/java/com/bookreview/bookreview/Model/UserDatas.java
package com.bookreview.bookreview.Model;

import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Data
@Document (collection = "UserDatas")
public class UserDatas 
{
    @Id
    private int id;

    private String uname,name,bio,gender,region,email,password;
    private boolean admin;

    private Images profilePic;
    private FavGenres intrestedGenres;

    private List<Reviews> reviews;
    private List<Notifications> notification;

}
